### DISCUSSION TITLE
Blank frame when using displacement map on gif

### DISCUSSION THREAD

**Discussion by hellocatfood** [ImageMagick/ImageMagick#2192] [Help] [answered]
I have the following input images. I'm using this map:

![ripples_flux_anim](https://user-images.githubusercontent.com/151947/85343152-0a812380-b4e4-11ea-8d4d-e79f141c9c57.gif)

To displace this image.

![gradient](https://user-images.githubusercontent.com/151947/85343165-10770480-b4e4-11ea-84f3-347cb7aadafb.png)

I'm using the following script to generate an animation, which works except it leaves a blank image at the end of the animation
```
convert \
	-dispose background \
	\( gradient.png null: \) \
	\( displacement_map.gif null: \) \
	-compose displace \
	-define compose:args=10x40 \
	-layers composite \
	output.gif
```
Which outputs this:

![output](https://user-images.githubusercontent.com/151947/85343208-33a1b400-b4e4-11ea-80c2-4ed490fbe220.gif)

Why does this blank image appear at the end and how can it be gotten rid of? I'm running Imagemagick 6.9.10 on Ubuntu 20.04.

**Comment by fmw42** [comment, marked as answer]
You have too many null: in your command and do not need the parentheses. Only one null: is used and that should be between the gradient and the displacement images. The last null: you used is adding an empty frame. See https://imagemagick.org/Usage/anim_mods/#composite

Try

```
convert \
-dispose background \
gradient.png null: \
displacement_map.gif \
-compose displace \
-define compose:args=10x40 \
-layers composite \
output.gif

```


![output](https://user-images.githubusercontent.com/13458218/85345640-cec86880-b4a7-11ea-93a6-c9f34f263d23.gif)
